### PR TITLE
Added a .windows platform

### PR DESCRIPTION
**Pull request by gmittert**

### CODE REVIEW

**Comment by gmittert** [comment]
Should this be behind some kind of version check?

File: Sources/TSCUtility/Platform.swift [hunk 1]
```diff
@@ -25,6 +26,9 @@ public enum Platform {
     public static var currentPlatform = Platform.findCurrentPlatform()
     /// Attempt to match `uname` with recognized platforms.
     private static func findCurrentPlatform() -> Platform? {
+      #if os(Windows)
+        return .windows
+      #else
         guard let uname = try? Process.checkNonZeroExit(args: "uname").spm_chomp().lowercased() else { return nil }
```
Comment (compnerd) [inline Sources/TSCUtility/Platform.swift]: Is there a reason to dynamically do this check? @aciidb0mb3r

Comment (aciidb0mb3r) [reply Sources/TSCUtility/Platform.swift]: Not that I know of. I don't remember why we have the API and maybe can get rid of it?

File: Sources/PackageDescription4/SupportedPlatforms.swift [hunk 2]
```diff
@@ -33,6 +33,9 @@ public struct Platform: Encodable {
 
     /// The Linux platform.
     public static let linux: Platform = Platform(name: "linux")
+
+    /// The Windows platform
+    public static let windows: Platform = Platform(name: "windows")
```
Comment (aciidb0mb3r) [inline Sources/PackageDescription4/SupportedPlatforms.swift]: We'll need to gate this using the package description availability. I suggest proposing this change separately. I don't think we need a proposal for this but I'll check with some other SwiftPM contributors.

Comment (gmittert) [reply Sources/PackageDescription4/SupportedPlatforms.swift]: Proposed separately in https://github.com/apple/swift-package-manager/pull/2382

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@compnerd is this still relevant given your recent Windows work?

**Comment by gmittert** [comment]
This one is still relevant. Platform detection as is on Windows won't work unless the windows machine happens to have "uname" in the path, which is possible, but not always the case. I'll also repeat the question above of if there is any particular reason it's currently done at runtime. Linux, yes, to distinguish the distros, but darwin could be behind a `#if os`.

File: Sources/TSCUtility/Platform.swift (hunk 1 above)
Comment (abertelrud) [reply Sources/TSCUtility/Platform.swift:32]: I don't see a reason for this either; it seems to me that `uname` should always return "Darwin" if and only if `os(Darwin)` is true.

**Comment by tomerd** [comment]
> This one is still relevant. Platform detection as is on Windows won't work unless the windows machine happens to have "uname" in the path, which is possible, but not always the case. I'll also repeat the question above of if there is any particular reason it's currently done at runtime. Linux, yes, to distinguish the distros, but darwin could be behind a #if os.

there seems to be at least some overlap with work @compnerd did on other PRs. @compnerd can you help coordinate this change with yours and get this in?

**Comment by compnerd** [comment]
@gmittert could you please rebase this?

File: Sources/TSCUtility/Platform.swift (hunk 1 above)
Comment (abertelrud) [reply Sources/TSCUtility/Platform.swift:32]: What I didn't realize when I wrote my previous comment was that there isn't actually a `os(Darwin)`, and I don't think that spelling out the current set of Darwin-based OSs is the right approach.  `canImport(Darwin)` would work, but I'd be a little nervous about assuming that only Darwin platforms have the Darwin module.  So maybe best to just ignore my comment and keep calling `uname` for Darwin.

**Comment by shahmishal** [comment]
We moved our default branch to `main` and deleted `master` branch, GitHub automatically closed the PR. Please update the base branch to `main`.  More detail about the branch update - https://forums.swift.org/t/updating-branch-names/40412

**Pull request closed without merging**